### PR TITLE
Fix for 64bit Windows - RuntimeError buffer has unexpected item byte size (8, expected 4) #4751

### PR DESCRIPTION
**Pull request by mdehoon**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Hmm, that doesn't looks to have solved everything, and sadly looks like it triggers some new doctest failures, e.g.

```
----------------------------------------------------------------------
File "C:\Py\Lib\site-packages\Bio\Align\__init__.py", line 1091, in Bio.Align.Alignment.parse_printed_alignment
Failed example:
    coordinates
Expected:
    array([[ 0,  1,  4,  6, 11, 12],
           [ 0,  1,  4,  4,  9,  9],
           [ 0,  0,  3,  5, 10, 11]])
Got:
    array([[ 0,  1,  4,  6, 11, 12],
           [ 0,  1,  4,  4,  9,  9],
           [ 0,  0,  3,  5, 10, 11]], dtype=int32)
----------------------------------------------------------------------
Ran 554 tests in 501.516 seconds
FAILED (failures = 3)
```

I was hoping we could remain agnostic on numpy v1 versus v2, maybe we still can.

**Comment by mdehoon** [comment]
@peterjc   The remaining AppVeyor failure is when showing a numpy array on a 32-bits Windows machine. The code works correctly, but `dtype=int64` is shown explicitly for this platform.

**Review by peterjc — APPROVED** [review]
If the problem was inconsistent size of int on the different platforms, explicitly using a 64 bit int makes sense. We may want to skip or tweak those doctests... eg cast to Python int for display?

**Comment by mdehoon** [comment]
> We may want to skip or tweak those doctests... eg cast to Python int for display?

Skipping makes more sense. If we cast, then the example becomes weird, as usually we would not cast.

**Comment by peterjc** [comment]
There is casting in the code snippet, which I agree may undermine the readability.

Or, there is casting in the code, so that on output the coordinates are Python ints rather than NumPy ints - the pragmatic solution we generally ended up with on #4676 for dealing with the NumPy 2 scalar repr changes.

I have not looked at how easy either might be in this case.

**Comment by mdehoon** [comment]
> Or, there is casting in the code, so that on output the coordinates are Python ints rather than NumPy ints

We cannot cast to Python ints, as the coordinates are an array. We can only cast the array of 64-bit integers to an array of 32-bit integers on 32-bit platforms. However, that is wasteful. Also, in extreme cases it can give an overflow error (for example, the genome of Paris japonica consists of 150 billion base pairs divided over 40 chromosomes, so 3.75 billion base pairs per chromosome on average, which is more than the largest 32-bit signed integer). So I think we need to stick to 64-bit integers. Then on 32-bit platforms the doctests will fail because `dtype=int64` is shown explicitly on 32-bit machines. But the code works correctly.

Quoting from https://numpy.org/devdocs/numpy_2_0_migration_guide.html :
> The default integer used by NumPy is now 64bit on all 64bit systems (and 32bit on 32bit system).

File: Doc/Tutorial/chapter_align.rst [hunk 1]
```diff
@@ -859,7 +859,9 @@ alignments:
 
 Now, let’s look at these two alignments:
 
-.. cont-doctest
+FIXME .. cont-doctest
+
+.. doctest ../Tests lib:numpy
```
Comment (peterjc) [inline Doc/Tutorial/chapter_align.rst:864]: Needs to go back to cont-doctest

**Comment by peterjc** [comment]
I like the print(array) solution to the doctests 👍

**Pull request merged**